src/condor_schedd.V6/schedd.cpp

```diff
@@ -16228,6 +16231,16 @@ Scheduler::sendAlives( int /* timerID */ )
 				daemonCore->Send_Signal( srec->pid, SIGKILL );
 			}
 		}
+
+		// If the match rec is merely claimed/idle, there is no Job
+		// or shadow
+		if (mrec->status == M_CLAIMED) {
+			int lease_duration = 6 * alive_interval;
+			if ((mrec->last_alive + lease_duration) < now ) {
+				mrec->needs_release_claim = false;
+				DelMrec( mrec );
+			} 
+		}
```

Comment:
This should probably not be done if m_startd_sends_alives=false. But rather than worry about that, we should just remove that code path entirely.